Formula/libirecovery.rb

```diff
@@ -0,0 +1,36 @@
+class Libirecovery < Formula
+  desc "Library to communicate with iDevices restore and recovery"
+  homepage "https://www.libimobiledevice.org/"
+
+  stable do
+    url "https://github.com/libimobiledevice/libirecovery/archive/0.1.1.tar.gz"
+    sha256 "81aab17730ebbbe598f91e8dd4249b76c1f0d6782c60061e59d3c52637b5305c"
+  end
+
+  head do
+    url "https://git.libimobiledevice.org/libirecovery.git"
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+    depends_on "libxml2"
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "libplist"
+  depends_on "libtasn1"
+  depends_on "libusbmuxd"
+  depends_on "openssl@1.1"
+
+  def install
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+
+    system "make", "install"
+  end
+
+  test do
+    system "irecovery", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hey - thank you for the advise

Sadly any tests would require a iPhone or other iDevice connected in DFU or recovery modes...

Reply:
@rickmark the only test case I could think of, which again is not a meaningful test in terms of usb connectivity at all, would be to query the device database and compare the result

Reply:
doing `idevice_id` exercises usb bus - even if there is nothing.  maybe we can reach the same point?

Reply:
Just keep in mind that something failing as expected is also a good validation of the homebrew install.